### PR TITLE
Maz style updates

### PR DESCRIPTION
**Pull request by mazhermon**
## Summary of changes
- responsive changes for better viewing on small screens
- add branding with colours and logo
- add new dark mode colour scheme (see new vars in CSS for values)
- allow full page to scroll for now and question to be asked at the bottom. Assume user will need to read the latest reply and scroll anyway. Simplifies scrolling on small screens and avoids double scroll bars. (Can look at other solutions in future if you'd prefer to keep that input always visible)
- make button text consistent - always says 'ask'
- indent chatbot responses to differentiate further between user input and bot responses
- auto focus the input on page load
- focus and hover states etc for all interactive element

TODO

### Sources
- could put those sources in an accordion to stop the unintentional hover on them as things load. User could click to open if interested to see them.
- could mark the sources up as a list and space items out a bit for easier hover
- could offer a touch accessible version of the sources hover data

### CODE REVIEW

File: assets/style.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/* TODO // once linked from head correctly, move all css in to here */
+body {
+    border: 4px solid tomato !important; /* remove me */
```
Comment (mazhermon) [inline assets/style.css:3]: this is just a test style to show that's working.
Would be good to move all styles out here soon

File: templates/index.html [hunk 2]
```diff
@@ -4,46 +4,264 @@
 <head>
     <meta charset="UTF-8">
     <title>Question bot</title>
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Noto+Sans:wght@300;400;500;700&family=Open+Sans:wght@300;400;600&display=swap" rel="stylesheet">
     <link href="https://unpkg.com/tailwindcss@^2/dist/tailwind.min.css" rel="stylesheet">
+    <!-- TODO: move styles here => <link rel="stylesheet" href="../assets/style.css"> -->
```
Comment (mazhermon) [inline templates/index.html:11]: wasn't sure how to get python local server to serve these up in the expected place, but something like this should be good

File: templates/index.html [hunk 3]
```diff
@@ -181,22 +399,43 @@
...
...h86.09q-37.87,119.85-166.68,119.84-48.9-.69-90.57-21.35t-66.46-64.4q-24.79-43.72-24.8-101.59,0-54.4,25.14-99.18t66.81-66.81a189.26,189.26,0,0,1,89.88-22Zm75.08,149.46q-6.89-39.94-25.48-57.86t-53-17.9q-35.82,0-56.48,20.31t-24.1,55.45Zm256.22-275.5V726.91h-97.81V235.14Zm277.57,126q73.7,0,117.78,51.66t44.08,139.81q0,80.6-42.36,132.25t-114,51.65q-70.26,0-106.76-53.72h-1.38V852.27h-97.8V370.82h93v45.46h1.38Q2165,361.19,2236,361.18ZM2129.9,549.9q0,52.35,22,82.65t63.37,30.31q42.71,0,63.71-30.31t21-82.65q0-53.73-22.72-84.37t-63.37-30.65q-40.63,0-62.33,30.3t-21.7,84.72ZM2621,361.18q50.28,0,89.54,23.42t61.64,67.84q22.39,44.43,22.39,102.28,0,5.5-.69,17.91H2537q1.38,42.71,22.38,66.46t64.4,23.77A94,94,0,0,0,2673,649.43q22.38-13.44,28.58-32.72h86.1Q2749.78,736.56,2621,736.55q-48.9-.69-90.57-21.35T2464,650.8q-24.79-43.72-24.8-101.59,0-54.4,25.14-99.18t66.81-66.81a189.33,189.33,0,0,1,89.89-22Zm75.07,149.46q-6.89-39.94-25.48-57.86t-53-17.9q-35.8,0-56.47,20.31T2537,510.64Zm363.67-149.46q12.39,0,21.35,3.44v90.92A165.41,165.41,0,0,0,3046,452.1q-96.42,0-96.42,114.33V726.91h-97.81V370.82h93v66.12h1.37a120.6,120.6,0,0,1,45.81-55.1q31.34-20.65,67.84-20.66Z" transform="translate(-81.08 -81.08)"/></g></svg>
```
Comment (mazhermon) [inline templates/index.html:404]: would be good to get these SVG's out into separate files in the future, maybe if you go to React or something later on

File: assets/style.css (hunk 1 above)
Comment (mrmattwright) [reply assets/style.css:3]: Deffo!

**Pull request merged**